doc: guidelines: update coding guidelines docs (#75315)
zephyrproject-rtos/zephyr: doc/contribute/coding_guidelines/index.rst

    Coding guideline rules may be removed/changed at any time by filing a
    GH issue/RFC.

.. important::

    **Current stage**
    The marked current **stage II** has not completely started yet. The prerequisites for this stage are

>     in progress. The tooling is in evaluation, CI setup and enforcement strategy is being worked on.

Reviewer: This is confusing. 

What stage are we in? Stage II is marked as the current stage, but this seems to claim we are still in Stage I.

Reviewer 2: I was thinking exactly the same, this should either be dropped entirely if we're happy enough with saying Stage II is the current one, even if there's some caveats. Or this should be reworked by marking "Stage I (COMPLETED)", leaving "Stage II" untouched, and wording your _important_ note along the lines of

```suggestion
.. important::

    **Current stage**
    The prerequisites for entering Stage II are currently being looked at: the tooling is in evaluation, CI setup and enforcement strategy is being worked on.
```

maybe add links to a couple relevant github issues if any to allow people to follow the progress

Author: You both are right. While writing it, it made perfect sense 😄.
@kartben I like the approach with the stage II entering prerequisites, I will adept it to it.
